delete-review-app/action.yml

```diff
@@ -0,0 +1,140 @@
+    required: false
+  global-config:
+    description: Environment for global config parameters
+    default: review
+    required: false
+  pr-number:
+    description: PR number for the review app
+    required: true
+  service-name:
+    description: name of the service
+    required: true
+  service-pretty:
+    description: pretty name of the service
+    required: true
+  service-short:
+    description: short name of the service
+    required: true
+  terraform-base:
+    description: Path to the terraform files
+    required: false
+    default: "terraform/application"
+  terraform-destroy-command:
+    description: terrafom destroy command in makefile
+    required: false
+    default: "terraform-destroy"
+
+runs:
+  using: composite
+  steps:
+    - name: Checkout
+      uses: actions/checkout@v4
+
+    - name: Set terraform version
+      id: set_env_var
+      shell: bash
+      run: |
+        terraform_version=$(awk '/{/{f=/^terraform/;next}f' ${{ inputs.terraform-base }}/terraform.tf | grep -o [0-9\.]*)
+        echo "TERRAFORM_VERSION=$terraform_version" >> $GITHUB_ENV
+
+    - name: Use Terraform ${{ env.TERRAFORM_VERSION }}
+      uses: hashicorp/setup-terraform@v3
+      with:
+        terraform_version: ${{ env.TERRAFORM_VERSION }}
+        terraform_wrapper: false
+
+    - uses: azure/login@v2
+      with:
+        creds: ${{ inputs.azure-credentials }}
+
+    - uses: google-github-actions/auth@v2
+      if: ${{ inputs.gcp-wip != '' }}
+      with:
+        project_id: ${{ inputs.gcp-project-id }}
+        workload_identity_provider: ${{ inputs.gcp-wip }}
+
+    - uses: DFE-Digital/github-actions/set-kubelogin-environment@master
+      with:
+        azure-credentials: ${{ inputs.azure-credentials }}
+
+    - name: Terraform Destroy
+      shell: bash
+      run: |
+        make ci ${{ inputs.environment }}  ${{ inputs.terraform-destroy-command  }}
+      env:
+        PR_NUMBER: ${{ inputs.pr-number }}
+
+    - name: Post Pull Request Comment
+      if: ${{ inputs.github-event-name == 'pull_request' }}
+      uses: marocchino/sticky-pull-request-comment@v2
+      with:
+        header: aks
+        message: |
+          Review app ${{ inputs.service-pretty }} deployed to <https://${{ inputs.service-name }}-pr-${{ inputs.pr-number }}.test.teacherservices.cloud> was deleted
+
+    - name: Install Storage Blob Extension
+      shell: bash
+      run: az extension add --name storage-blob-preview
+
+    - name: Set environment variables
+      shell: bash
+      run: |
+        source global_config/${{ inputs.global-config }}.sh
+        echo "RESOURCE_GROUP_NAME=${AZURE_RESOURCE_PREFIX}-${{ inputs.service-short }}-${CONFIG_SHORT}-rg" >> $GITHUB_ENV
+        echo "STORAGE_ACCOUNT_NAME=${AZURE_RESOURCE_PREFIX}${{ inputs.service-short }}${CONFIG_SHORT}tfsa" >> $GITHUB_ENV
+        echo "CONTAINER_NAME=terraform-state" >> $GITHUB_ENV
+        echo "TF_STATE_FILE=pr-${{ inputs.pr-number }}_kubernetes.tfstate" >> $GITHUB_ENV
```

Comment:
I can see us needing to be able to override container_name and tf_state_file for a lot of the existing services. Which might be easier if generated from the calling service?

Reply:
implemented